### PR TITLE
Mct7367-tests

### PR DESCRIPTION
**Pull request by unlikelyzero**
### Describe your changes:
Tests for the #7367 feature

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [ ] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

**Comment by unlikelyzero** [comment]
All that's left is role=progressbar

File: src/ui/components/ProgressBar.vue [hunk 1]
```diff
@@ -25,6 +25,10 @@
       class="c-progress-bar__bar"
       :class="{ '--indeterminate': !progressPerc }"
       :style="styleBarWidth"
+      role="progressbar"
+      :aria-valuenow="progressPerc"
```
Comment (ozyx) [inline src/ui/components/ProgressBar.vue:29]: woah neat

**Review by ozyx — APPROVED** [review]
Awesome ! I just had one review comment from a pending review, and this needs a good linting. Should be g2g after that

File: e2e/tests/functional/plugins/inspectorDataVisualization/numericData.e2e.spec.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ test.describe('Testing numeric data with inspector data visualization (i.e., dat
     await page.goto('./', { waitUntil: 'domcontentloaded' });
   });
 
-  test('Can click on telemetry and see data in inspector', async ({ page, context }) => {
+  test('Can click on telemetry and see data in inspector @2p', async ({ page, context }) => {
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/inspectorDataVisualization/numericData.e2e.spec.js:39]: driveby for @scottbell

File: e2e/tests/functional/plugins/notebook/notebookSnapshotImage.e2e.spec.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+import fs from 'fs/promises';
+import { fileURLToPath } from 'url';
+
+import { createDomainObjectWithDefaults } from '../../../../appActions.js';
+import { expect, test } from '../../../../pluginFixtures.js';
+
+const NOTEBOOK_NAME = 'Notebook';
+
+test.describe('Snapshot image tests', () => {
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/notebook/notebookSnapshotImage.e2e.spec.js:35]: broke this out into its own suite for speed

File: e2e/tests/functional/plugins/notebook/notebookSnapshots.e2e.spec.js [hunk 4]
```diff
@@ -91,22 +85,13 @@ test.describe('Snapshot Container tests', () => {
 
     await page.getByLabel('Take a Notebook Snapshot').click();
     await page.getByRole('menuitem', { name: 'Save to Notebook Snapshots' }).click();
-    await page.getByRole('button', { name: 'Show' }).click();
+    await page.getByLabel('Show Snapshots').click();
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/notebook/notebookSnapshots.e2e.spec.js:88]: locator fixes

File: e2e/tests/visual-a11y/components/header.visual.spec.js [hunk 5]
```diff
@@ -50,6 +50,17 @@ test.describe('Visual - Header @a11y', () => {
       scope: header
     });
   });
+
+  test('show snapshot button', async ({ page, theme }) => {
```
Comment (unlikelyzero) [inline e2e/tests/visual-a11y/components/header.visual.spec.js:54]: realized we were missing this one

File: e2e/tests/visual-a11y/components/header.visual.spec.js [hunk 6]
```diff
@@ -50,6 +50,17 @@ test.describe('Visual - Header @a11y', () => {
...
+    await page.getByLabel('Take a Notebook Snapshot').click();
+
+    await page.getByRole('menuitem', { name: 'Save to Notebook Snapshots' }).click();
+
+    await percySnapshot(page, `Notebook Snapshot Show button (theme: '${theme}')`, {
+      scope: header
+    });
+    await expect(await page.getByLabel('Show Snapshots')).toBeVisible();
```
Comment (unlikelyzero) [inline e2e/tests/visual-a11y/components/header.visual.spec.js:62]: need to wait for button to appear in header

File: e2e/tests/functional/plugins/importAndExportAsJSON/exportAsJson.e2e.spec.js [hunk 7]
```diff
@@ -33,24 +33,21 @@ import {
...
-    // });
-    // const page = await context.newPage();
-
-    // Navigate to the page
-    await page.goto('./');
-
+  let folder;
+  test.beforeEach(async ({ page }) => {
+    // Go to baseURL
+    await page.goto('./', { waitUntil: 'networkidle' });
```
Comment (ozyx) [inline e2e/tests/functional/plugins/importAndExportAsJSON/exportAsJson.e2e.spec.js:40]: networkidle?

File: e2e/tests/functional/plugins/notebook/notebookSnapshotImage.e2e.spec.js (hunk 3 above)
Comment (ozyx) [reply e2e/tests/functional/plugins/notebook/notebookSnapshotImage.e2e.spec.js:35]: doesn't it parallelize by file and not suite? or has that changed

Comment (unlikelyzero) [reply e2e/tests/functional/plugins/notebook/notebookSnapshotImage.e2e.spec.js:35]: it's by file. I broke up the image and regular snapshot suites
